### PR TITLE
fix(aws-kms): Incomplete KMS Resource Policy Permissions

### LINKED ISSUE #3458: KMS generates incomplete IAM resource policy
* **I'm submitting a ...**
  - [x] :beetle: bug report
  - [ ] :rocket: feature request
  - [ ] :books: construct library gap
  - [ ] :phone: security issue or vulnerability => Please see [policy](https://github.com/aws/aws-cdk/security/policy)
  - [ ] :question: support request => Please see note at the top of this template.

* **What is the current behavior?**
*If the current behavior is a :beetle:bug:beetle:: Please provide the steps to reproduce*

When creating a **Customer Managed KMS Key** the generated resource policy for the account principal is incomplete. At least when it comes to S3 buckets.

**Create KMS Key and bucke with KMS enabled**
```typescript
const kmsKey = new kms.Key(this, 'KeyName', {
      enableKeyRotation: true,
      enabled: true,
      removalPolicy: cdk.RemovalPolicy.DESTROY
    });

new s3.Bucket(this, 'MyBucket', {
      encryption: s3.BucketEncryption.KMS,
      encryptionKey: kmsKey,
      removalPolicy: cdk.RemovalPolicy.DESTROY
    });
```
With this default configuration the key generated is missing an action for to be able to write to S3 buckets (read seems to not be affected). With the **cli**, **sdk** or **console** no create (put or otherwise) on the bucket is allowed.

The error given through the aws cli is **"An error occurred (AccessDenied) when calling the PutObject operation: Access Denied"**

The fix according to this [AWS support article](https://aws.amazon.com/premiumsupport/knowledge-center/s3-access-denied-error-kms/) is to add the following to the actions statement **kms:GenerateDataKey**.

* **What is the expected behavior (or behavior of feature suggested)?**

Expected behaviour is that write operations to S3 buckets should work.

* **What is the work around if any?**

The work around is to add the following to the CDK code.

```typescript
kmsKey.addToResourcePolicy(
      new iam.PolicyStatement({
        actions: ['kms:GenerateDataKey'],
        principals: [new iam.AccountPrincipal(cdk.Aws.ACCOUNT_ID)],
        resources: ["*"]
      })
    );
```

or this, which is more precise

```typescript
kmsKey.addToResourcePolicy(
      new iam.PolicyStatement({
        actions: ['kms:GenerateDataKey'],
        principals: [new iam.AccountRootPrincipal()],
        resources: ['*']
      })
    );
```

This completes the resource policy and write operations to S3 now work.

* **Please tell us about your environment:**

  - CDK CLI Version: 1.2.0
  - Module Version: ?
  - OS: [all]
  - Language: [Typescript ]

### PR DESCRIPTION
**Pull request by yandy-r**
Fixes #3458 where incomplete default resource policy for root account principal was generated and requiring a workaround.

See issue #3458 for the complete reference.

----

Please read the [contribution guidelines](https://github.com/aws/aws-cdk/blob/master/CONTRIBUTING.md) and follow the pull-request checklist.

*By submitting this pull request, I confirm that my contribution is made under the terms of the Apache-2.0 license*

### CODE REVIEW

**Comment by eladb** [comment]
Build is failing...

**Comment by IPyandy** [comment]
> Build is failing...

yea unfortunately I keep finding these dependencies only after the build fails once pushed. A local run of build+test pass.

if there's something I'm missing for this to be found locally I'm all ears.

right now I'm doing a search for anything in the `@aws-cdk` folder with those key actions to hopefully find them all.

**Comment by RomainMuller** [comment]
@IPyandy you might want to make a full-clean build, by running `git clean -fqdX` before re-running `./build.sh`...

File: packages/@aws-cdk/aws-codepipeline-actions/test/integ.cfn-template-from-repo.lit.expected.json [hunk 1]
```diff
@@ -59,10 +60,7 @@
               "Effect": "Allow",
               "Principal": {
                 "AWS": {
-                  "Fn::GetAtt": [
-                    "PipelineRoleD68726F7",
-                    "Arn"
-                  ]
+                  "Fn::GetAtt": ["PipelineRoleD68726F7", "Arn"]
```
Comment (eladb) [inline packages/@aws-cdk/aws-codepipeline-actions/test/integ.cfn-template-from-repo.lit.expected.json]: What's going on here? Why did the formatting change? These expectation files are supposed to be formatted automatically when running "cdk run integ".

Please revert any changes to the expectation files were not produced by this change. It's important so we can review.

Comment (IPyandy) [reply packages/@aws-cdk/aws-codepipeline-actions/test/integ.cfn-template-from-repo.lit.expected.json]: My bad there, I thought I turned off when saving these files, should be back now.

**Comment by IPyandy** [comment]
This one single line of code change has turned out to be quite a pain. This kms:GenerateDataKey action trickled into quite a few different packages. I did not expect so many dependencies for it.

Local code builds from scratch 100% now, hopefully, the codebuild and travisci stages pass now.

**Comment by eladb** [comment]
Thanks for this @IPyandy

**Pull request merged**